connectors/aws/aws-textract/src/main/java/io/camunda/connector/textract/caller/PollingTextractCaller.java

```diff
@@ -6,82 +6,89 @@
-      final String jobId = result.getJobId();
-      final TextractTask firstTextractTask = prepareTextractTask(jobId, null, textractClient);
-      final GetDocumentAnalysisResult firstDocumentResult =
-          executeTask(firstTextractTask, 0, executorService);
-
-      allBlocks = new ArrayList<>(firstDocumentResult.getBlocks());
-      lastDocumentResult = firstDocumentResult;
-      String nextToken = firstDocumentResult.getNextToken();
-
-      while (StringUtils.isNoneEmpty(nextToken)) {
-        final TextractTask nextTextractTask = prepareTextractTask(jobId, nextToken, textractClient);
-        GetDocumentAnalysisResult nextDocumentResult =
-            executeTask(nextTextractTask, DELAY_BETWEEN_POLLING, executorService);
-        nextToken = nextDocumentResult.getNextToken();
-        allBlocks.addAll(nextDocumentResult.getBlocks());
-        lastDocumentResult = nextDocumentResult;
-      }
-    }
+    LOGGER.debug("Started document analysis with jobId: {}", jobId);
 
-    lastDocumentResult.setBlocks(allBlocks);
-    return lastDocumentResult;
-  }
+    GetDocumentAnalysisResult firstResult = pollUntilComplete(jobId, textractClient);
+
+    List<Block> allBlocks = new ArrayList<>(firstResult.getBlocks());
+    GetDocumentAnalysisResult lastResult = firstResult;
+    String nextToken = firstResult.getNextToken();
 
-  private TextractTask prepareTextractTask(
-      String jobId, String nextToken, AmazonTextract textractClient) {
-    GetDocumentAnalysisRequest documentAnalysisReq =
-        new GetDocumentAnalysisRequest().withJobId(jobId).withMaxResults(MAX_RESULT);
+    while (StringUtils.isNotEmpty(nextToken)) {
+      GetDocumentAnalysisRequest nextRequest =
+          new GetDocumentAnalysisRequest()
+              .withJobId(jobId)
+              .withMaxResults(MAX_RESULT)
+              .withNextToken(nextToken);
 
-    if (StringUtils.isNoneEmpty(nextToken)) {
-      documentAnalysisReq.withNextToken(nextToken);
+      GetDocumentAnalysisResult nextResult = textractClient.getDocumentAnalysis(nextRequest);
+      nextToken = nextResult.getNextToken();
+      allBlocks.addAll(nextResult.getBlocks());
+      lastResult = nextResult;
     }
 
-    return new TextractTask(documentAnalysisReq, (AmazonTextractAsync) textractClient);
+    lastResult.setBlocks(allBlocks);
+    return lastResult;
   }
 
-  private GetDocumentAnalysisResult executeTask(
-      TextractTask task, long delay, ScheduledExecutorService executorService) throws Exception {
-    ScheduledFuture<GetDocumentAnalysisResult> nextDocumentResultFuture =
-        executorService.schedule(task, delay, SECONDS);
-    return nextDocumentResultFuture.get();
+  private GetDocumentAnalysisResult pollUntilComplete(String jobId, AmazonTextract textractClient)
+      throws InterruptedException {
+
+    final long INITIAL_DELAY_MS = 5000;
+    final long MAX_DELAY_MS = 30 * 60 * 1000;
```

Comment:
30 min sounds like a lot! I think we can safely poll every couple of minutes, this would provide faster reaction time, wdyt?